<launch>
  <arg name="init_x"/>
  <arg name="init_y"/>
  <arg name="init_z"/>
  <arg name="map_size_x"/>
  <arg name="map_size_y"/>
  <arg name="map_size_z"/>
  <arg name="quadrotor_odometry_topic"/>
  <arg name="payload_odometry_topic"/>

<node pkg="nodelet" type="nodelet" name="manager" args="manager" output="screen"/>
  <node pkg="nodelet" type="nodelet" name="so3_quadrotor"
    args="load so3_quadrotor/Nodelet manager" output="screen">
    <param name="init_x" value="$(arg init_x)"/>
    <param name="init_y" value="$(arg init_y)"/>
    <param name="init_z" value="$(arg init_z)"/>

    <rosparam file="$(find uav_simulator)/config/so3_quadrotor.yaml"/>
    <remap from="~odom"         to="$(arg quadrotor_odometry_topic)"/>
    <remap from="~odom_payload" to="$(arg payload_odometry_topic)"/>
    <remap from="~imu"          to="imu" />
    <remap from="~imu_payload"  to="imu_payload" />
    <remap from="~so3cmd"       to="so3cmd"/>
    <remap from="~vis"          to="vis"/>
  </node>

  <!-- <node pkg="impact_control" type="mpc_controller_node" name="mpc_controller_node" output="screen">
      <rosparam command="load" file="$(find impact_control)/config/mpc.yaml" />
      <rosparam command="load" file="$(find impact_control)/config/model.yaml" />
      <remap from="~odom"                         to="$(arg quadrotor_odometry_topic)"/>
      <remap from="~payload_odom"                 to="$(arg payload_odometry_topic)"/>
      <remap from="/mavros/imu/data"              to="imu"/>
      <remap from="~payload_imu/data"             to="imu_payload"/>
	  <remap from="~traj"                         to="planning/trajectory"/>
      <remap from="/mavros/setpoint_raw/attitude" to="so3cmd"/>
      <remap from="~start_trigger"                to="/move_base_simple/goal"/>
      <remap from="~cmd_trigger"                  to="/hummingbird/goal"/>
    </node> -->

  <node pkg="odom_visualization" name="odom_visualization" type="odom_visualization" output="screen">
        <remap from="~odom" to="$(arg payload_odometry_topic)"/>
        <param name="color/a"          value="1.0"/>    
        <param name="color/r"          value="0.0"/>        
        <param name="color/g"          value="0.0"/>        
        <param name="color/b"          value="0.0"/>       
        <param name="covariance_scale" value="100.0"/>       
        <param name="robot_scale"      value="1.0"/>
        <param name="tf45"             value="false"/>
        <param name="drone_id"         value="1"/>
  </node> 
  <node pkg="local_sensing_node" type="pcl_render_node" name="pcl_render_node" output="screen">
        <rosparam command="load" file="$(find local_sensing_node)/params/camera.yaml" />
        <param name="sensing_horizon"  value="15.0" />
        <param name="sensing_rate"     value="30.0"/>
        <param name="estimation_rate"  value="30.0"/>

        <param name="map/x_size"     value="$(arg map_size_x)"/>
        <param name="map/y_size"     value="$(arg map_size_y)"/>
        <param name="map/z_size"     value="$(arg map_size_z)"/>

        <remap from="~global_map"              to="/env_generator/global_cloud"/>
        <remap from="~odometry"                to="$(arg quadrotor_odometry_topic)"/>
        <remap from="~pcl_render_node/cloud"   to="pcl_render_node/cloud"/>
  </node>
</launch>
